start.py

```diff
@@ -724,6 +760,99 @@ def _configure_kudu(deployment, cluster, kudu_version):
                                                                configs)
 
 
+def _setup_ssl(cluster, product_list, ssl_authentication_required):
+    _setup_ssl_ca_authority(cluster)
+    if not ssl_authentication_required:
+        for product in product_list:
+            _setup_ssl_encryption(cluster, product)
+    else:
+        for product in product_list:
+            _setup_ssl_encryption_authentication(cluster, product)
+
+
+def _setup_ssl_ca_authority(cluster):
+    ssl_ca_authority_commands = [
+        'mkdir -p {}'.format(SSL_SERVER_CONTAINER_DIR),
+        'mkdir -p {}'.format(SSL_CLIENT_CONTAINER_DIR),
+        'cd {}'.format(SSL_SERVER_CONTAINER_DIR),
+        ('openssl req -new -newkey rsa:4096 -days 365 -x509 -subj "/CN=Kafka-Security-CA"'
+         ' -keyout ca-private-key -out ca-public-key -nodes'),
+    ]
+    cluster.primary_node.execute(' && '.join(ssl_ca_authority_commands))
+
+
+def _setup_ssl_encryption(cluster, product):
+    cur_product = product.lower()
+    product_server_keystore = '{}.{}'.format(cur_product, SSL_SERVER_KEYSTORE)
+    product_server_truststore = '{}.{}'.format(cur_product, SSL_SERVER_TRUSTSTORE)
+
+    # Configuration of server side e.g. for Kafka broker.
```

Comment:
You call the variable `ssl_kafka_broker_commands` but these kinda look more general than that? Or is it Kafka specific? If it is, you'd probably want to add a `if service == 'KAFKA'` or something to ensure it doesn't get applied if support for other services is added in eventually.

Reply:
Done.